report/map_keys.go

```diff
@@ -0,0 +1,69 @@
+	"docker_container_id":              "docker_container_id",
+	"docker_container_ips":             "docker_container_ips",
+	"docker_container_ips_with_scopes": "docker_container_ips_with_scopes",
+	"docker_container_name":            "docker_container_name",
+	"docker_container_network_mode":    "docker_container_network_mode",
+	"docker_container_restart_count":   "docker_container_restart_count",
+	"docker_container_state":           "docker_container_state",
+	"docker_container_state_human":     "docker_container_state_human",
+	"docker_container_uptime":          "docker_container_uptime",
+	"docker_exec_container":            "docker_exec_container",
+	"docker_image_id":                  "docker_image_id",
+	"docker_image_name":                "docker_image_name",
+	"docker_image_size":                "docker_image_size",
+	"docker_image_virtual_size":        "docker_image_virtual_size",
+	"docker_is_in_host_network":        "docker_is_in_host_network",
+	"docker_pause_container":           "docker_pause_container",
+	"docker_remove_container":          "docker_remove_container",
+	"docker_restart_container":         "docker_restart_container",
+	"docker_start_container":           "docker_start_container",
+	"docker_stop_container":            "docker_stop_container",
+	"docker_unpause_container":         "docker_unpause_container",
+	"does_not_make_connections":        "does_not_make_connections",
+	"host":                                "host",
+	"host_node_id":                        "host_node_id",
+	"kubernetes_available_replicas":       "kubernetes_available_replicas",
+	"kubernetes_created":                  "kubernetes_created",
+	"kubernetes_desired_replicas":         "kubernetes_desired_replicas",
+	"kubernetes_fully_labeled_replicas":   "kubernetes_fully_labeled_replicas",
+	"kubernetes_ip":                       "kubernetes_ip",
+	"kubernetes_labels_name":              "kubernetes_labels_name",
+	"kubernetes_labels_pod-template-hash": "kubernetes_labels_pod-template-hash",
+	"kubernetes_name":                     "kubernetes_name",
+	"kubernetes_namespace":                "kubernetes_namespace",
+	"kubernetes_observed_generation":      "kubernetes_observed_generation",
+	"kubernetes_replicas":                 "kubernetes_replicas",
+	"kubernetes_scale_down":               "kubernetes_scale_down",
+	"kubernetes_scale_up":                 "kubernetes_scale_up",
+	"kubernetes_strategy":                 "kubernetes_strategy",
+	"kubernetes_unavailable_replicas":     "kubernetes_unavailable_replicas",
+	"kubernetes_updated_replicas":         "kubernetes_updated_replicas",
+	"name":              "name",
+	"pid":               "pid",
+	"pod":               "pod",
+	"port":              "port",
+	"ppid":              "ppid",
+	"procspied":         "procspied",
+	"reverse_dns_names": "reverse_dns_names",
+	"snooped_dns_names": "snooped_dns_names",
+	"threads":           "threads",
+}
```

Comment:
I suggest you use the defined constants here. That will also identify entries in the above which are no longer present in current reports (`conntracked`, `procspied` comes to mind).

Reply:
That seems to end up with a rather ugly mish-mash of imports from all over Scope plus a few keys that are runtime-generated.

Also I don't think we want to reduce the performance on older reports every time we remove keys from newer ones.  It costs basically nothing to have one extra string in this list, for a given order of magnitude.

Reply:
I am struggling to work out how this list is maintained. Or indeed how I would go about reviewing its contents as part of reviewing this PR. How do we make sure it contains all keys we want? How do we make sure it contains no spurious keys (e.g. because of typos etc)? My suggestion to use the constants was aimed at (imperfectly) addressing that.

Reply:
Perhaps if I checked in the code that creates the list?

Currently that doesn't exist as one thing - it went via some Go code, some awk code, some regexp-replace, etc.